### PR TITLE
Convert to proper actions, add build workflow, upload action unit tests, 0.1.3

### PR DESCRIPTION
**Pull request by Vainock**
### Description
- Adds Unit Tests for the Upload Action
- Improves and refines the tests for the Download Action
- Converts the actions to proper actions
- Removes the dist.js files and adds a workflow to automatically build and commit the dist.js to the main branch on pushes
- Improves existing code in both actions
- Transforms the build command into its own script to provide compatibility with other OS other than Windows
- Updates all packages
- Removes unnecessary package versioning
- Shows a more verbose Jest output
- Checks existance of submodules and translation directories to prepare for submodule removal

Btw, I have also resolved https://github.com/obsproject/crowdin-synchronization/projects/1#card-74947220
I didn't start working on https://github.com/obsproject/crowdin-synchronization/projects/1#card-74947210 as I see no point when the submodules get merged into the main repository very soon anyway.

### Motivation and Context
- https://github.com/obsproject/crowdin-synchronization/projects/1#column-16571839
- Unit tests are great to verify functionality.
- My drive to improve this project in any way possible.

### How Has This Been Tested?
Using `npm run test` and a fork to verify the build workflow works.

### Types of changes

- New feature (non-breaking change which adds functionality)
- Tweak (non-breaking change to improve existing functionality)
- Performance enhancement (non-breaking change which improves efficiency)
- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: src/download.ts [hunk 1]
```diff
@@ -133,6 +128,9 @@ export async function getSourceFiles(filePaths: Map<number, string>): Promise<Ma
 			}
 			currentFileStrings!.set(sourceString.identifier, sourceString.text as string);
 		}
+		if (projectStrings.length !== 500) {
+			break;
+		}
```
Comment (WizardCM) [inline src/download.ts:133]: What's the meaning behind this special number `500`?

Comment (Vainock) [reply src/download.ts:133]: The max amount of items (in this case strings) you can retrieve from the API is 500. What I do here is check if I have reached that limit in my request and if yes send another request with an offset of 500.

Comment (WizardCM) [reply src/download.ts:133]: Ah, that makes sense. Could you set a variable `MAX_API_PAGESIZE` elsewhere and use it here instead to make it more readable?

**Review by WizardCM — CHANGES_REQUESTED** [review]
Hmm, the Jest tests seem be less verbose? And they're passing when they should likely fail:

0.1.2:
```
> jest

PASS tests/download.test.ts
  ✓ getLanguages (27 ms)
  ✓ getFilePaths (15 ms)
  ✓ getTranslators (67 ms)
  ✓ buildProject (29 ms)
  ✓ getSourceFiles (43 ms)
  ✓ processBuild (41 ms)
  ✓ localeFile (16 ms)
  ✓ desktopFile (1 ms)

Test Suites: 1 passed, 1 total
Tests:       8 passed, 8 total
Snapshots:   0 total
Time:        4.263 s
Ran all test suites.
```

0.1.3:
```
> jest

PASS tests/download.test.ts
PASS tests/upload.test.ts

Test Suites: 2 passed, 2 total
Tests:       9 passed, 9 total
Snapshots:   0 total
Time:        4.779 s
Ran all test suites.
Notice: UI/data/locale/en-US.ini updated on Crowdin.
Notice: plugins/sndio/data/locale/en-US.ini updated on Crowdin.
Error: Some files couldn't be found on Crowdin and/or their export paths need to be fixed to get updated:
 plugins/decklink/data/locale/en-US.ini
```

**Comment by Vainock** [comment]
1. The output is less verbose because Jest only shows all tests if you only run a single test file, I'm now running 2 files. This is only a visual thing.
2.
```
Notice: UI/data/locale/en-US.ini updated on Crowdin.
Notice: plugins/sndio/data/locale/en-US.ini updated on Crowdin.
Error: Some files couldn't be found on Crowdin and/or their export paths need to be fixed to get updated:
 plugins/decklink/data/locale/en-US.ini
```
is completely intended and even the error is expected. I've now found a way to hide the ouput during tests.
I use these console messages to check if everything works as intended:
```ts
expect(noticeMock).toBeCalledWith('UI/data/locale/en-US.ini updated on Crowdin.');
expect(noticeMock).toBeCalledWith('plugins/sndio/data/locale/en-US.ini updated on Crowdin.');
expect(errorMock).toBeCalledWith("Some files couldn't be found on Crowdin and/or their export paths need to be fixed to get updated:\n plugins/decklink/data/locale/en-US.ini");
```

**Pull request merged**